src/app.ts

```diff
@@ -248,7 +248,7 @@ export class App extends Construct {
   private writeConstructMetadata(apiObjects: ApiObject[]) {
     const resources: { [key: string]: any } = {};
     for (const apiObject of apiObjects) {
-      resources[apiObject.name] = { path: apiObject.node.path };
+      resources[apiObject.name] = { path: apiObject.node.path, Kind: apiObject.kind };
```

Comment:
I can see how the error:

```console
error: unable to decode "dist\\construct-metadata.json": Object 'Kind' is missing in
```

led you to this solution. But its actually not the right one. The error happens because `kubectl` attempts to apply the `construct-metadata.json` to the cluster, but this file is not a manifest, and so should be ignored, and does therefore not need to have `Kind`.

This file should only be created if a user explicitly sets `recordConstructMetadata` on the `App`. But it looks like it is created by default. So actually we just need to fix this [issue](https://github.com/cdk8s-team/cdk8s-core/issues/1042).

However, we will still have a problem in case `recordConstructMetadata` is set to true. We should investigate how to create that file in a way that is ignored by `kubectl`. Can you open a different issue on that?